spec/unit/provider/login_spec.rb

```diff
@@ -0,0 +1,171 @@
+################################################################################
+# (C) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+require 'spec_helper'
+require_relative '../../../lib/puppet/provider/login.rb'
+
+describe 'login', unit: true do
+  context 'given the OneView authentication' do
+    let(:auth_settings) do
+      {
+        url:              'https://172.16.100.185',
+        token:            'NzA3OTg2NjM2NTk21xd-TkijbSwOjm2AvXDAL4LPG49D9K8u',
+        user:             'administrator',
+        password:         'secret123',
+        ssl_enabled:      true,
+        api_version:      300,
+        log_level:        'debug',
+        hardware_variant: 'Synergy'
+      }
+    end
+
+    let(:json_file) do
+      File.read('spec/support/fixtures/unit/provider/login_oneview.json')
+    end
+
+    it 'should load configuration from file' do
+      allow(ENV).to receive(:[]).with('ONEVIEW_AUTH_FILE').and_return(nil)
```

Comment:
Should just set the variables, not mock them.
`ENV[ONEVIEW_AUTH_FILE] = nil`
Same for the mocks bellow.